webclient/doi/model_serializers.py

```diff
@@ -0,0 +1,994 @@
+                    {
+                        "relationType": "IsPublishedIn",
+                        "relatedIdentifier": f"{settings.DIGIARCHIV_URL}{externi_odkaz.externi_zdroj.ident_cely}",
+                        "resourceTypeGeneral": externi_odkaz.externi_zdroj.typ.heslar_odkaz.kod,
+                        "relatedIdentifierType": "URL",
+                    }
+                )
+        return related_identifiers
+
+    def _serialize_related_items(self):
+        related_items = []
+        for externi_odkaz in self._get_externi_odkaz_query():
+            externi_odkaz: ExterniOdkaz
+            externi_zdroj = externi_odkaz.externi_zdroj
+            if externi_zdroj.stav == EZ_STAV_POTVRZENY:
+                related_item = {
+                    "relatedItemType": externi_zdroj.typ.heslar_odkaz.kod,
+                    "relationType": "IsPublishedIn",
+                    "relatedItemIdentifier": {
+                        "relatedItemIdentifier": externi_zdroj.doi or externi_zdroj.ident_cely,
+                        "relatedItemIdentifierType": "DOI",
+                    },
+                    "creators": [
+                        serialize_creator_contributor(ea.autor) for ea in externi_zdroj.externizdrojautor_set.all()
+                    ],
+                    "titles": externi_zdroj.nazev,
+                }
+                if externi_zdroj.sbornik_nazev and not externi_zdroj.casopis_denik_nazev:
+                    related_item["volume"] = externi_zdroj.sbornik_nazev
+                elif externi_zdroj.casopis_denik_nazev and not externi_zdroj.sbornik_nazev:
+                    related_item["volume"] = externi_zdroj.casopis_denik_nazev
+                if externi_zdroj.casopis_rocnik and not externi_zdroj.datum_rd:
+                    related_item["issue"] = externi_zdroj.casopis_rocnik
+                elif not externi_zdroj.casopis_rocnik and externi_zdroj.datum_rd:
+                    related_item["issue"] = externi_zdroj.datum_rd
+                if externi_zdroj.vydavatel and not externi_zdroj.organizace:
+                    related_item["publisher"] = externi_zdroj.vydavatel
+                elif not externi_zdroj.vydavatel and externi_zdroj.organizace:
+                    related_item["publisher"] = externi_zdroj.organizace
+                related_item["publicationYear"] = externi_zdroj.rok_vydani_vzniku
+                related_item["edition"] = externi_zdroj.edice_rada
+                related_item["contributors"] = (
+                    [serialize_creator_contributor(ea.editor) for ea in externi_zdroj.externizdrojeditor_set.all()],
+                )
+                related_items.append(related_item)
+        return related_items
+
+    def _serialize_subjects(self):
+        serialized_subjects = super()._serialize_subjects()
+        try:
+            if self.record.archeologicky_zaznam and self.record.archeologicky_zaznam.stav == AZ_STAV_ARCHIVOVANY:
```

Comment:
Téhle podmínce nerozumím - proč tu je? Pokud tato funkce je pro lokalitu, pak to nedává smysl, ale možná to špatně chápu.

Reply:
Opraveno.